src/provider.py

```diff
@@ -175,40 +144,32 @@ def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
             event.defer()
             return
 
-        if (
-            event.relation.app != self.charm.app or not self.charm.app.planned_units() == 0
-        ):  # avoid on own charm during teardown
-            provider_relation_config = self.provider_relation_config(event=event)
+        if event.relation.app != self.charm.app or not self.charm.app.planned_units() == 0:
 
             self.kafka_auth.load_current_acls()
-
+            username = f"relation-{event.relation.id}"
             self.kafka_auth.remove_all_user_acls(
-                username=provider_relation_config["username"],
+                username=username,
             )
-            self.kafka_auth.delete_user(username=provider_relation_config["username"])
-
+            self.kafka_auth.delete_user(username=username)
             # non-leader units need cluster_config_changed event to update their super.users
-            self.charm.model.get_relation(PEER).data[self.charm.app].update(
-                {provider_relation_config["username"]: ""}
-            )
+            self.charm.model.get_relation(PEER).data[self.charm.app].update({username: ""})
 
     def update_connection_info(self):
```

Comment:
This should probably get a type-hint.

Reply:
I think this function should stay. This methods is called when something changed in the configuration of the charm (this function is trigger on several events: ). So it is responsible to update the some parameters (e.g., zookeeper-uris, bootstrap-server and tls configuration) in response of the events handled by the `on_config_changed` function.

Reply:
Yes, sorry, my mistake. I thought we had similar method in the KafkaProvides class with similar business logic, but that's quite not the case. Good to go as it. 

Yes, following @marcoppenheimer suggestion, I'd also possibly just put a "-> None", as it would be best to be explicit even when with don't return anything